.github/workflows/python-code-generation.yaml

```
@@ -0,0 +1,75 @@
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Python code generation
+
+on: [push]
```

Set filter to master branch
```suggestion
on:
  push:
    branches:
      - master
```

---

I would rather have this workflow trigger on a push over any branch.
It would ensure any change to the `.proto` files can be compiled by protoc.
Otherwise, the risk is to push on master without knowing if the changes to the files are ok.

---

Thanks to caching this workflow takes just a couple of seconds on each run most of the times.